src/daemon/daemon.cpp

```diff
@@ -1627,59 +1627,33 @@ try // clang-format on
     mpl::ClientLogger<InfoReply, InfoRequest> logger{mpl::level_from(request->verbosity_level()), *config->logger,
                                                      server};
     InfoReply response;
+    InstanceSnapshotsMap instance_snapshots_map;
 
     // Need to 'touch' a report in the response so formatters know what to do with an otherwise empty response
     request->snapshot_overview() ? (void)response.mutable_snapshot_overview()
                                  : (void)response.mutable_detailed_report();
     bool have_mounts = false;
     bool deleted = false;
-    auto fetch_instance_info = [&](VirtualMachine& vm) {
-        const auto& name = vm.vm_name;
-        auto info = response.mutable_detailed_report()->add_details();
-        auto instance_info = info->mutable_instance_info();
-        auto present_state = vm.current_state();
-        info->set_name(name);
-        if (deleted)
-        {
-            info->mutable_instance_status()->set_status(mp::InstanceStatus::DELETED);
-        }
-        else
-        {
-            info->mutable_instance_status()->set_status(grpc_instance_status_for(present_state));
-        }
 
-        auto vm_image = fetch_image_for(name, config->factory->fetch_type(), *config->vault);
-        auto original_release = vm_image.original_release;
-
-        if (!vm_image.id.empty() && original_release.empty())
-        {
-            try
-            {
-                auto vm_image_info = config->image_hosts.back()->info_for_full_hash(vm_image.id);
-                original_release = vm_image_info.release_title.toStdString();
-            }
-            catch (const std::exception& e)
-            {
-                mpl::log(mpl::Level::warning, category, fmt::format("Cannot fetch image information: {}", e.what()));
-            }
-        }
+    auto populate_snapshot_fundamentals = [&](std::shared_ptr<const Snapshot> snapshot, auto& fundamentals) {
```

Comment:
This function is getting really big, with all the sub functions and all. I think this would be a good time to extract the individual parts into either private methods or unnamed namespace helpers.